src/index.ts

```
@@ -12,8 +12,35 @@ export class UserAPI {
     this.users = initial_users;
   }
 
-  getAllUsers(): ReadonlyArray<User> {
-    const users_array = Object.values(this.users);
-    return users_array;
+  getAllUsers(): Record<string, User> {
+    return this.users;
   }
-}
+
+  addUser(userObj: User) {
+    const newUserId = (Math.floor(Math.random() * 900)).toString();
+    // const newUserId = v4();
+    const new_user = {
+      ...userObj,
+      id: newUserId
+    }
+    return {
+      ...this.users,
+      [newUserId]: new_user
+    }
+  }
+
+  getUserById(id: string) {
+    return (
+      this.users[id]
+    )
+  }
+
+  deleteUserById(id: string) {
+    delete this.users[id];
+    if(this.users[id]) {
+      return this.users[id]
+    } else {
+      return null
```

nice catch on handling the 'user does not exist' condition.  The upcoming error handling ticket will update this function to throw an error, rather than return null.

---

addressed